File: TheaterSuggestions/Java/ScreeningSeatingArrangementSuggestionsInfra/src/main/java/org/octo/seatingplacessuggestions/infra/AuditoriumDto.java
package org.octo.seatingplacessuggestions.infra;



import java.util.List;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonProperty;

public record AuditoriumDto(@JsonProperty("Rows") Map<String, List<SeatDto>> rows, @JsonProperty("Corridors") List<CorridorDto> corridors) {
}

File: TheaterSuggestions/CSharp/ScreeningSeatingArrangementSuggestions.Infra/AuditoriumLayoutRepository/AuditoriumDto.cs
using ExternalDependencies.AuditoriumLayoutRepository;

namespace ScreeningSeatingArrangementSuggestions.Infra.Adapter.AuditoriumLayoutRepository;

public class AuditoriumDto
{
    public Dictionary<string, IReadOnlyList<SeatDto>> Rows { get; }
    public IEnumerable<CorridorDto> Corridors { get; }
}